packages/@tinacms/cli/src/cmds/compile/index.ts

```diff
@@ -41,74 +44,107 @@ export const resetGeneratedFolder = async () => {
 
 export const compile = async (_ctx, _next) => {
   logger.info(logText('Compiling...'))
-  // FIXME: This assume it is a schema.ts file
-  if (
-    !fs.existsSync(tinaPath) ||
-    !fs.existsSync(path.join(tinaPath, 'schema.ts'))
-  ) {
+  let schemaExists = true
+  try {
+    getSchemaPath({ projectDir: tinaPath })
+  } catch (_) {
+    // getSchemaPath will throw an error if it is not found
+    schemaExists = false
+  }
+  if (!schemaExists) {
     // The schema.ts file does not exist
     logger.info(
       dangerText(`
       .tina/schema.ts not found, Creating one for you...
       See Documentation: https://tina.io/docs/tina-cloud/cli/#getting-started"
       `)
     )
+    // We will default to TS?
     const file = path.join(tinaPath, 'schema.ts')
     // Ensure there is a .tina/schema.ts file
     await fs.ensureFile(file)
     // Write a basic schema to it
     await fs.writeFile(file, defaultSchema)
   }
 
-  // Turn the TS files into JS files so they can be exacted
-  await transpile(tinaPath, tinaTempPath)
+  // Turns the schema into JS files so they can be run
+  try {
+    await transpile(tinaPath, tinaTempPath)
+  } catch (e) {
+    throw new BuildSchemaError(e)
+  }
 
   // Delete the node require cache for .tina temp folder
   Object.keys(require.cache).map((key) => {
     if (key.startsWith(tinaTempPath)) {
       delete require.cache[require.resolve(key)]
     }
   })
-
-  const schemaFunc = require(path.join(tinaTempPath, 'schema.js'))
-  const schemaObject: TinaCloudSchema = schemaFunc.default
-  await fs.outputFile(
-    path.join(tinaConfigPath, 'schema.json'),
-    JSON.stringify(schemaObject, null, 2)
-  )
-  await fs.remove(tinaTempPath)
+  try {
+    const schemaFunc = require(path.join(tinaTempPath, 'schema.js'))
+    const schemaObject: TinaCloudSchema = schemaFunc.default
+    await fs.outputFile(
+      path.join(tinaConfigPath, 'schema.json'),
+      JSON.stringify(schemaObject, null, 2)
+    )
+    await fs.remove(tinaTempPath)
```

Comment:
related issue here as above - the temp file won't get cleaned up if an error is thrown. you can consider using a `finally` block, though I'm not completely clear on whether `finally` will execute when you re-throw from `catch`

Reply:
Yeah I _dont think_ it would. I will do some refactoring to account for this.

Reply:
@dwalkr I updated the code to account for this